### PR TITLE
10 course functionality

### PR DESCRIPTION
**Pull request by reesedychiao**

### CODE REVIEW

File: frontend/src/app/components/Assignments.tsx
Comment (n30w) [inline frontend/src/app/components/Assignments.tsx:1]: This is a good start, but try and break it down even more. What you can try to do is to separate the upload button/box, assignment selection box, and other components as you see fit. Remember separation of concerns. In regard to sharing state between components, like getting the value of one and putting it into the other, you can lookup NextJS or React child to parent component data passing.

File: frontend/src/app/components/Courses.tsx [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
 import Image from "next/image";
 
 interface CourseProps {
   coursename: string;
   professor: string;
   loc: string;
+  onClick?: () => void;
 }
 
 const Course: React.FC<CourseProps> = (props) => {
```
Comment (n30w) [inline frontend/src/app/components/Courses.tsx:13]: the React.FC<params> from TypeScript is good and appropriate here. Make sure to refactor the rest of your component code to follow this syntax and consistency. For example, the Assignments component.

File: frontend/src/app/course/[id]/[id].tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
+"use client";
```
Comment (n30w) [inline frontend/src/app/course/[id]/[id].tsx:1]: We don't need to use 'use client' here right? Leaving this as a React Server Component (RSC) is good enough, isn't it?

File: frontend/src/app/components/Discussions.tsx [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
       </h2>
-      <div>
-        <Image src={""} alt={""} />
-        <h3 className="text-sm text-gray-400 mb-2">
-          Neo Alabastro posted on Feb 9, 2024 9:16 PM
-        </h3>
-      </div>
-      <p className="text-sm font-light">In my experience, ...</p>
+      <h3 className="text-white">
+        Neo Alabastro posted on Feb 9, 2024 9:16 PM
```
Comment (n30w) [inline frontend/src/app/components/Discussions.tsx:10]: The date part of this string can be turned into a date component.

**Review by n30w — CHANGES_REQUESTED** [review]
For page.tsx files, please use "export default function [function name, like Page or Home]() {}" syntax instead of fat arrow syntax "page: any = () => {}". Fat arrow can be used for everything except page.tsx.

**Comment by reesedychiao** [comment]
> For page.tsx files, please use "export default function function name, like Page or Home {}" syntax instead of fat arrow syntax "page: any = () => {}". Fat arrow can be used for everything except page.tsx.

This raises an error though. I think typescript requires you to specify the type and that requires a fat arrow function?

File: frontend/src/app/components/Courses.tsx (hunk 1 above)
Comment (reesedychiao) [reply frontend/src/app/components/Courses.tsx:13]: I already declared the interface for other components similar to this one, is there something I need to add?

Comment (n30w) [reply frontend/src/app/components/Courses.tsx:13]: No, I was just pointing out that the React.FC<> syntax is good.

Comment (reesedychiao) [reply frontend/src/app/components/Courses.tsx:13]: okay nice!

**Comment by n30w** [comment]
> This raises an error though. I think typescript requires you to specify the type and that requires a fat arrow function?

Strange, it doesn't raise an error for me. I'll change it to mine and we can look at it in person.

**Review by n30w — APPROVED** [review]
Thanks for the work @reesedychiao

**Pull request merged**